spec/chef/server_spec.rb

```diff
@@ -11,9 +11,12 @@
   end
 
   it "determines the server version" do
-    s = Chef::Server.new("http://api.example.com")
-    allow(s).to receive(:open).and_return(StringIO.new("Chef Server 1.8.1\nother stuff\nother stuff"))
-    expect(s.version.to_s).to eq("1.8.1")
+    s1 = Chef::Server.new('http://api.example.com')
+    s2 = Chef::Server.new('http://api.example.com')
+    allow(s1).to receive(:open).and_return(StringIO.new("Chef Server 1.8.1\nother stuff\nother stuff"))
+    expect(s1.version.to_s).to eq('1.8.1')
+    allow(s2).to receive(:open).and_return(StringIO.new("Package: chef-server/chef-server-nginx/12.17.42/20180413212943\nother stuff\nother stuff"))
+    expect(s2.version.to_s).to eq('12.17.42')
```

Comment:
minor: mind breaking this into another case with a description that mentions this is what we get for a habitat based package.